clients/src/main/java/org/apache/kafka/common/metrics/Sensor.java

```diff
@@ -250,7 +250,16 @@ public void checkQuotas() {
         checkQuotas(time.milliseconds());
     }
 
-    public void checkQuotas(long timeMs) {
+    public synchronized void checkQuotas(long timeMs) {
+        checkQuotasInternal(timeMs);
+    }
+
+    // visible for testing
+    protected void checkQuotasNonThreadSafe(long timeMs) {
```

Comment:
Can we make it package-private instead of protected?

Reply:
Done.